lib/ansible/modules/identity/keycloak/keycloak_user.py

```diff
@@ -0,0 +1,441 @@
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: keycloak_user
+
+short_description: Allows administration of Keycloak users via Keycloak API
+
+version_added: "2.9"
+
+description:
+    - This module allows the administration of Keycloak clients via the Keycloak REST API. It
+      requires access to the REST API via OpenID Connect; the user connecting and the client being
+      used must have the requisite access rights. In a default Keycloak installation, admin-cli
+      and an admin user would work, as would a separate client definition with the scope tailored
+      to your needs and a user having the expected roles.
+
+    - The names of module options are snake_cased versions of the camelCase ones found in the
+      Keycloak API and its documentation at U(https://www.keycloak.org/docs-api/4.8/rest-api/index.html/).
+      Aliases are provided so camelCased versions can be used as well. If they are in conflict
+      with ansible names or previous used names, they will be prefixed by "keycloak".
+
+options:
+    state:
+        description:
+            - State of the user
+            - On C(present), the user will be created (or updated if it exists already).
+            - On C(absent), the user will be removed if it exists
+        choices: [ present, absent ]
+        default: present
+        type: str
+
+    realm:
+        description:
+            - The realm to create the user in.
+        default: master
+        type: str
+
+    attributes:
+        description:
+            – a dictionary with the key and the value to put in keycloak.
+            Keycloak will always return the value in a list of one element.
+            Keys and values are converted into string.
+        required: false
+        type: dict
+
+    user_id:
+        description:
+            - user_id of client to be worked on. This is usually an UUID. This and I(client_username)
+              are mutually exclusive.
+        aliases: [ userId ]
+        type: str
+
+    keycloak_username:
+        description:
+            - username of user to be worked on. This and I(user_id) are mutually exclusive.
+            - keycloak lower the username
+        aliases: [ keycloakUsername ]
+        type: str
+
+    email_verified:
+        description:
+            - show if the user email have been verified
+        required: false
+        type: bool
+        aliases: [ emailVerified ]
+
+    enabled:
+        description:
+            - show if the user can logged in
+        required: false
+        type: bool
+
+    email:
+        description:
+            - the user email
+            - this module does not check the validity of the email
+            - when using the api, there is no check about the validity of the email in keycloak
+            - but with manual action, the format is checked
+        required: false
+        type: str
+
+    required_actions:
+        description:
```

Comment:
This could use more explanation.

Reply:
Updated in ff2053a46a9548ab87e89bf785602931997ed205.